### PR TITLE
[Glossary] Terminology definition for testing per issue #1495

### PR DESCRIPTION
**Pull request by trevgc**
Added section 1.7 for test related terminology and updated with existing text in general terminology section this is test related and with definition text from related issue.

### CODE REVIEW

File: doc/tech/glossary.md [hunk 1]
```diff
@@ -146,15 +146,24 @@ Terms not defined by Kubernetes:
 - Observability = Observability is a measure of how well internal states of a system can be inferred from knowledge of its external outputs. ([Wikipedia:Observability](https://en.wikipedia.org/wiki/Observability))
 
 <a name="1.7"></a>
-## 1.7 Other Referenced Terminology
+## 1.7 Test Related Terminology
+
+- **Performance Measurement:** The procedure or set of operations having the object of determining a Measured Value or Measurement Result. In this context, Performance Measurements reflect data generated and collected within the cloud infrastructure, that reflects a perfromance aspect of the cloud infrastructure. For example, a count of frames or packets traversing an interface, memory usage information, other resource usage and availability, etc. These data may be instantaneous or accumulated, and made available (i.e. exposed) based on permissions and contexts (e.g., workload vs. infra)
```
Comment (pgoyal01) [inline doc/tech/glossary.md:151]: nitpick: Since using "For example" then "etc." is unnecessary

File: doc/tech/glossary.md [hunk 2]
```diff
@@ -146,15 +146,24 @@ Terms not defined by Kubernetes:
...
+
+- **Performance Testing**: The main objective of CNTT performance testing is to understand if a potential BOM is functioning correctly to get expected performance. It requires a set of performance testing tools (open source) that help with the dimensioning of a solution by an operator. This testing provides a comparison between the SUT and a CNTT reference implementation. Performance testing for comparisons between commercial implementations is not a goal of CNTT. Performance testing relies on established benchmark specifications.
```
Comment (pgoyal01) [inline doc/tech/glossary.md:153]: Acronyms --first use (e.g., BOM)

IMO, the testing doesn't provide a comparison -- the results of the testing can be used for comparison

File: doc/tech/glossary.md (hunk 1 above)
Comment (trevgc) [reply doc/tech/glossary.md]: Added period

File: doc/tech/glossary.md (hunk 2 above)
Comment (trevgc) [reply doc/tech/glossary.md]: Agree, made the change as suggested

File: doc/tech/glossary.md [hunk 3]
```diff
@@ -146,15 +146,24 @@ Terms not defined by Kubernetes:
...
+
+- **Performance Testing**: The main objective of CNTT performance testing is to understand if a potential BOM is functioning correctly to get expected performance. It requires a set of performance testing tools (open source) that help with the dimensioning of a solution by an operator. Testing results may be useful to compare infrastructure capabilites between the System Under Test (SUT) and a CNTT reference implementation of RA-2. Performance testing for the purpose of comparing between different commercial implementations is not a goal of CNTT. Performance testing relies on well established benchmark specifications.
```
Comment (acmacm) [inline doc/tech/glossary.md:153]: This section should become an introduction for the Test Related Terminology Section, so I suggest to delete the text here and move under the section title.
```suggestion
    
```

File: doc/tech/glossary.md [hunk 4]
```diff
@@ -146,15 +146,24 @@ Terms not defined by Kubernetes:
 - Observability = Observability is a measure of how well internal states of a system can be inferred from knowledge of its external outputs. ([Wikipedia:Observability](https://en.wikipedia.org/wiki/Observability))
 
 <a name="1.7"></a>
-## 1.7 Other Referenced Terminology
+## 1.7 Test Related Terminology
+
```
Comment (acmacm) [inline doc/tech/glossary.md:150]: New intro
```suggestion
The main objective of CNTT performance testing is to understand if a potential BOM is functioning correctly to get expected performance. It requires a set of performance testing tools (open source) that help with the dimensioning of a solution by an operator. Testing results may be useful to compare infrastructure capabilites between the System Under Test (SUT) and a CNTT reference implementation of RA-2. Performance testing for the purpose of comparing between different commercial implementations is not a goal of CNTT. Performance testing relies on well established benchmark specifications.
```

**Comment by rabi-abdel** [comment]
@trevgc would you mind please addressing comments above?

**Comment by pgoyal01** [comment]
Issue #1495

File: doc/tech/glossary.md (hunk 4 above)
Comment (pgoyal01) [reply doc/tech/glossary.md:150]: BOM -- acronym first use. If it is Bill of Materials then we can't say that "a potential BOM is functioning correctly..."

Comment (pgoyal01) [reply doc/tech/glossary.md:150]: Also, why does this require an operator? Could performance testing be conducted on a reference implementation stood up by a testing organisation?

**Comment by rabi-abdel** [comment]
@xmulligan shall this PR be closed?

**Comment by xmulligan** [comment]
@trevgc would you like to address the comments or should we close this PR pending further discussion in CNTT?

**Comment by xmulligan** [comment]
fixes #1495

**Comment by xmulligan** [comment]
The discussion will be moved to https://wiki.lfnetworking.org/display/LN/Performance+Testing+Minutes until a formal PR will be raised

File: doc/tech/glossary.md (hunk 4 above)
Comment (trevgc) [reply doc/tech/glossary.md:151]: This PR is for text in the glossary and to keep consistent I have not added an introduction. Most of the text is worked into the definitions now.

BOM and specific ref to operators removed.

File: doc/tech/glossary.md (hunk 3 above)
Comment (trevgc) [reply doc/tech/glossary.md]: Updated per CVC

File: doc/tech/glossary.md (hunk 4 above)
Comment (trevgc) [reply doc/tech/glossary.md:151]: Removed BOM and specific ref to operators

**Comment by trevgc** [comment]
Related
1. Issue ... [RC2 Ch02] Define terminology and scope of RC2 (Functional vs. Performance) #1495 
2. PR already merged ... [RC2 Ch01] Proposed text per issue #1495 re. terminology, scope (considering performance) #1642

**Review by xmulligan — APPROVED** [review]

File: doc/tech/glossary.md [hunk 5]
```diff
@@ -146,15 +147,30 @@ Terms not defined by Kubernetes:
...
+- **Calibration**: The process of checking and/or adjusting a stimulus generation or measurement device with a known reference value, to improve the overall quality of the measured results.  Calibration may be very simple, such as a comparison of the configured traffic generator sending rate and measured rate using a simple SUT (System Under Test)  such as loop-back cable between interfaces, such that that the known reference value is the published nominal interface rate.
```
Comment (tomkivlin) [inline doc/tech/glossary.md:152]: Duplicated "that"
```suggestion
- **Calibration**: The process of checking and/or adjusting a stimulus generation or measurement device with a known reference value, to improve the overall quality of the measured results.  Calibration may be very simple, such as a comparison of the configured traffic generator sending rate and measured rate using a simple SUT (System Under Test)  such as loop-back cable between interfaces, such that the known reference value is the published nominal interface rate.
```

File: doc/tech/glossary.md [hunk 6]
```diff
@@ -146,15 +147,30 @@ Terms not defined by Kubernetes:
...
+
+- **Reference Value**: A measured or established result or outcome for comparison with new measurements. For example, the reference value or expected outcome of a Functional Test is "PASS". The reference value or expected outcome of a Performance Measurement or Benchmarking test may be the value measured for the previous SUT release, or the published value or theoretical limit of a simple SUT.
+
+- **API Testing**: Testing against a protocol specification for conformance.
+
+- **Functional Testing**: The main objective of functional testing is the verification of a specific stimulus / response within the SUT, including the expected behavior.  These tests generally result in a binary outcome, i.e. pass / fail.  For example, verification of an "API call" and its associated response, such as the instantiation of a VM (or container) and verification of the VM's (or container's) existence (expected behavior), or the ability to activate a specific feature of the SUT (e.g. SR-IOV).
```
Comment (tomkivlin) [inline doc/tech/glossary.md:158]: "The main objective of functional testing is the verification of a specific stimulus / response within the SUT, including the expected behavior." doesn't feel specific enough for "functional testing" - meaning the same could be said of non-functional testing.  e.g. non-functional testing of scaling is the verification of a specific stimulus / response within the SUT (increased load/traffic), including the expected behaviour (system auto-scaled).  

I wonder if the first sentence could be made more specific, e.g.
```suggestion
- **Functional Testing**: The main objective of functional testing is the verification of compliance against specific functional requirements using a specific stimulus / response within the SUT, including the expected behavior.  These tests generally result in a binary outcome, i.e. pass / fail.  For example, verification of an "API call" and its associated response, such as the instantiation of a VM (or container) and verification of the VM's (or container's) existence (expected behavior), or the ability to activate a specific feature of the SUT (e.g. SR-IOV).
```

**Review by tomkivlin — CHANGES_REQUESTED** [review]
Couple of additional suggestions:
- Should we add "System Under Test" as a definition?
- Should we include any other non-functional testing (i.e. security, scalability, stress testing, etc.) or just performance testing?

**Comment by trevgc** [comment]
@tomkivlin , accepted your edits thanks. Re. 

> Couple of additional suggestions:
> 
> * Should we add "System Under Test" as a definition?
> * Should we include any other non-functional testing (i.e. security, scalability, stress testing, etc.) or just performance testing?

These and other definitions could be added. This pull request address specific test related terms that are a known source of confusion and were resolved through a number of discussions.

**Review by tomkivlin — APPROVED** [review]

**Comment by tomkivlin** [comment]
> @tomkivlin , accepted your edits thanks. Re. 
> 
> > Couple of additional suggestions:
> > 
> > * Should we add "System Under Test" as a definition?
> > * Should we include any other non-functional testing (i.e. security, scalability, stress testing, etc.) or just performance testing?
> 
> These and other definitions could be added. This pull request address specific test related terms that are a known source of confusion and were resolved through a number of discussions.
> 
> 

Understood. I've created an issue to handle these additions.

File: doc/tech/glossary.md (hunk 3 above)
Comment (taylor) [reply doc/tech/glossary.md]: @acmacm can this be resolved?

**Comment by taylor** [comment]
@trevgc please resolve the conflicts then we are ready to merge re: RC2 call 9/3

File: doc/tech/glossary.md (hunk 4 above)
Comment (pgoyal01) [reply doc/tech/glossary.md:151]: @trevgc @acm This Intro change has not been made

**Review by scottsteinbrueck — CHANGES_REQUESTED** [review]
@trevgc please resolve or remove Milestone=M3. If conflicts are resolved, also remove label=Conflict. thank you.

**Comment by collivier** [comment]
@xmulligan +1 to close it. We could reopen it later if needed.

**Comment by trevgc** [comment]
> @trevgc do you want to resolve the conflicts here or should we close it?

@xmulligan I just saw this and looked back at the history. The pull request was approved by 3 people, I am not sure what the conflict was that needed to be resolved? Lets not lose this as the glossary needs some improvements.

**Review by pgoyal01 — APPROVED** [review]
Subject to conflict resolution

**Comment by pgoyal01** [comment]
@trevgc It states that there are conflicts and that will prevent merge

**Comment by xmulligan** [comment]
@trevgc if you click on the resolve conflicts button, it will show what needs to be resolved

**Comment by trevgc** [comment]
Closed this pull request as it cannot be merged due to files moving around. Have replaced it with #2201

**Pull request closed without merging**